src/dstack/_internal/server/background/tasks/process_running_jobs.py

```diff
@@ -193,6 +209,48 @@ async def _process_job(job_id: UUID):
                         )
                         session.add(new_job_model)
                 # job will be terminated by process_finished_jobs
+
+        if (
+            initial_status != job_model.status
+            and job_model.status == JobStatus.RUNNING
+            and run.run_spec.configuration.type == "service"
+        ):
+            res = await session.execute(
+                select(GatewayModel).where(
+                    GatewayModel.project_id == project.id,
+                    GatewayModel.name == job.job_spec.gateway.gateway_name,
+                )
+            )
+            gateway = res.scalar_one()
```

Comment:
I think GatewayModel can be deleted while the service is running and and in this case res.scalar_one() will throw an error. We should either disallow deleting gateways with running services or handle the case when the gateway was deleted (e.g. fail the service).